src/uct/ib/rdmacm/rdmacm_cm.c

```diff
@@ -83,11 +83,33 @@ uct_rdmacm_cm_device_context_init(uct_rdmacm_cm_device_context_t *ctx,
     uint64_t general_obj_types_caps;
     ucs_status_t status;
     void *cap;
-    int ret;
 #endif
+    struct ibv_port_attr port_attr;
+    struct ibv_device_attr dev_attr;
+    int ret;
+    int i;
 
     ctx->num_dummy_qps = 0;
 
+    ret = ibv_query_device(verbs, &dev_attr);
+    if (ret != 0) {
+        ucs_error("ibv_query_device(%s) failed: %m", dev_name);
+        return UCS_ERR_IO_ERROR;
+    }
+
+    UCS_BITMAP_CLEAR(&ctx->is_eth);
+    for (i = 0; i < dev_attr.phys_port_cnt; ++i) {
+        ret = ibv_query_port(verbs, i + 1, &port_attr);
```

Comment:
can we define first port macro as 1, and use it also in https://github.com/openucx/ucx/blob/master/src/uct/ib/base/ib_device.c#L760